### PR TITLE
Numark Scratch:  Manual

### PR DESCRIPTION
**Pull request by NotYourAverageAl**
mapping PR  is [4834](https://github.com/mixxxdj/mixxx/pull/4834)

### CODE REVIEW

**Comment by NotYourAverageAl** [comment]
I closed this PR by mistake. I'll have a look over  the weekend. thanks

**Comment by JoergAtGithub** [comment]
I see that you had problems to get the SVG images through pre-commit. But changing the format from SVG to PNG is not the right approach.
Do you've pre-commit installed locally?

**Comment by NotYourAverageAl** [comment]
> I see that you had problems to get the SVG images through pre-commit. But changing the format from SVG to PNG is not the right approach. Do you've pre-commit installed locally?

Local pre-commit is failing as well. I have attached the files below if you want to have a look

numark_scratch
numark_scratch_performancepads

Optimize SVG files.......................................................Failed
- hook id: scour
- files were modified by this hook

Scour processed file "source/_static/controllers/numark_scratch.svg" in 4440 ms: 1664951/5531187 bytes new/orig -> 30.1%
Scour processed file "source/_static/controllers/numark_scratch_performancepads.svg" in 4523 ms: 1671409/5778175 bytes new/orig -> 28.9%
Using 16 concurrent jobs.

**Comment by NotYourAverageAl** [comment]
I created a new fresh svg and it seems local & online pre-commit worked this time around.

**Comment by ronso0** [comment]
Please add a white background to SVGs to ensure it's readable even if a browser uses different main background color (dark/night mode or other reason).

IMO you can then squash the SVG/png commits into one and force-push, if that's okay for all reviewers @JoergAtGithub ?

**Comment by ronso0** [comment]
strange, pre-commit complains again.
You can simply apply the patch from the check, lines 173-176
https://github.com/mixxxdj/manual/actions/runs/5333920658/jobs/9665031422?pr=492#step:4:173

**Comment by NotYourAverageAl** [comment]
I have another look when I get home. I think scour initially removed the white background when optimising. 🤷🏾‍♂️

**Comment by ronso0** [comment]
Okay, great, pre-commit is happy now!

Now please
> ... squash the SVG/png commits into one and force-push, if that's okay for all reviewers @JoergAtGithub ?

File: source/hardware/controllers/numark_scratch.rst [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+It's equipped with an InnoFader crossfader and 8 performance pads across 3 modes, and
+6 dedicated software FX selectors with dual toggle triggers. The Scratch is DVS ready under Mixxx
+
+-  `Manufacturer’s product page <https://www.numark.com/product/scratch>`__
+-  `Serato DJ Hardware Page <https://serato.com/dj/hardware/numark-scratch>`__
+-  `Owner’s Manual <https://www.numark.com/images/product_downloads/Scratch_-_User_Guide_-_v1.2.pdf>`__
+-  `Driver Download Page <https://www.numark.com/product/scratch>`__
+-  `Mapping Forum Thread <https://mixxx.discourse.group/t/numark-scratch-mapping/25186>`__
+
+.. versionadded:: 2.3.0
```
Comment (Swiftb0y) [inline source/hardware/controllers/numark_scratch.rst:16]:
```suggestion
.. versionadded:: 2.3.6
```

File: source/hardware/controllers/numark_scratch.rst [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+Drivers
+-------
+
+You can download the latest Windows driver from the `manufacturer’s website <https://www.numark.com/product/scratch>`__.
+The Scratch is class complaint on Mac.
```
Comment (Swiftb0y) [inline source/hardware/controllers/numark_scratch.rst:22]:
```suggestion
The Scratch is class complaint on Mac and Linux.
```

**Review by Swiftb0y — COMMENTED** [review]
lgtm otherwise

**Comment by JoergAtGithub** [comment]
> if that's okay for all reviewers @JoergAtGithub ?

I've not reviewed this PR - feel free to merge this, when you think it's ready

**Comment by ronso0** [comment]
I've started a text review, but had to leave. I hope to finish this asap

**Comment by NotYourAverageAl** [comment]
Please note I'll need to rebase this manual to 2.4 as the mapping is based on the 2.4 branch. Or the rebase the mapping back to 2.3 to get it out quicker.

**Comment by JoergAtGithub** [comment]
We updated the JavaScript engine from 2.3 to 2.4. Therefore you would need to change the syntax at various places to backport it to 2.3. Therefore better keep the mapping on target 2.4 and rebase the manual!

File: source/hardware/controllers/numark_scratch.rst [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+==========================  =====================
+Control                     Mode
+==========================  =====================
+:hwlabel:`PAD MODE` CUE     Hot Cue Mode
+:hwlabel:`PAD MODE` SAMPLE  Sampler Mode
+:hwlabel:`PAD MODE` ROLL    Loop Roll Mode
+==========================  =====================
+
+CUE mode: Pads 1-4 trigger Hotcue Cues, and if no Cue point is present on that pad/slot, will set a Cue point
+in the respective pad/slot. Holding SHIFT while pressing a pad will clear the Cue point from that pad/slot.
```
Comment (ronso0) [inline source/hardware/controllers/numark_scratch.rst:123]:
```suggestion
CUE mode: Pads 1-4 trigger the respective hotcues, and if no hotcue point is set on that pad, will set a Cue point
in the respective pad/slot. Holding SHIFT while pressing a pad will clear the Cue point from that pad.
```
```suggestion
CUE mode: Pads 1-4 trigger Hotcue Cues, and if no Cue point is present on that pad/slot, will set one.
in the respective pad/slot. Holding SHIFT while pressing a pad will clear the Cue point from that pad/slot.
```
You can also make the mode names bold with `**MODE**`

**Review by ronso0 — CHANGES_REQUESTED** [review]
Some ideas to improve the flow and reduce redundancy.

File: source/hardware/controllers/numark_scratch.rst [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+together in input channels 5-6, so Mixxx can record and broadcast them.
+
+The knobs for :hwlabel:`MASTER`, :hwlabel:`BOOTH`, :hwlabel:`MIC LEVEL`, :hwlabel:`MIC TONE` and :hwlabel:`CUE GAIN` are controlling the hardware mixer of the built-in audio interface.
+Hence, turning the knobs will not change values in the Mixxx :term:`GUI` and you’ll need to set the Mixxx knobs to their default values when using the controller:
+
+- Set the master/booth/headphones/microphone levels to 100% (knob center position)
+
+.. note::
+   You should assign the :guilabel:`Vinyl Control` input channels even if you do not intend to use timecode vinyl.
+   These channel assignments are also needed to make the :hwlabel:`CH 1/2 PC/LINE/PHONO` switches work (i. e. passing through line or phono input to the left/right deck).
```
Comment (ronso0) [inline source/hardware/controllers/numark_scratch.rst:55]: actually it was just "also" that made this a bit unclear to me
```suggestion
    The :hwlabel:`PC / LINE/PHONO` switches work only if the :guilabel:`Vinyl Control` input channels are assigned to software inputs. So even if you do not intend to use timecode vinyl but want to pass through line or phono inputs directly to the left/right deck, you need assign these channels in the Sound Hardware preferences.
```
Did I get that right? I.e. you refer to the hardware passthrough, not to the PASSTHROUGH option in Mixxx?

**Review by ronso0 — CHANGES_REQUESTED** [review]

File: source/hardware/controllers/numark_scratch.rst (hunk 4 above)
Comment (NotYourAverageAl) [reply source/hardware/controllers/numark_scratch.rst:55]: Yep your version sound clearer

**Comment by ronso0** [comment]
Thanks, LGTM now.

Waiting for the mapping PR

**Review by Swiftb0y — APPROVED** [review]

**Pull request merged**